GrpcDocStringExtractorTest sometimes fails

/cc @anuraaga 

Not sure when this happens but saw this failure a few times today:

```
com.linecorp.armeria.server.grpc.GrpcDocStringExtractorTest > enumType FAILED
    java.lang.AssertionError: 
    Expecting:
     <{}>
    to contain:
     <[MapEntry[key="armeria.grpc.testing.CompressionType", value=" Compression algorithms
    "]]>
    but could not find:
     <[MapEntry[key="armeria.grpc.testing.CompressionType", value=" Compression algorithms
    "]]>
        at com.linecorp.armeria.server.grpc.GrpcDocStringExtractorTest.enumType(GrpcDocStringExtractorTest.java:76)

com.linecorp.armeria.server.grpc.GrpcDocStringExtractorTest > method FAILED
    java.lang.AssertionError: 
    Expecting:
     <{}>
    to contain:
     <[MapEntry[key="armeria.grpc.testing.TestService/UnaryCall", value=" One request followed by one response.
    "]]>
    but could not find:
     <[MapEntry[key="armeria.grpc.testing.TestService/UnaryCall", value=" One request followed by one response.
    "]]>
        at com.linecorp.armeria.server.grpc.GrpcDocStringExtractorTest.method(GrpcDocStringExtractorTest.java:40)

com.linecorp.armeria.server.grpc.GrpcDocStringExtractorTest > nestedMessage FAILED
    java.lang.AssertionError: 
    Expecting:
     <{}>
    to contain:
     <[MapEntry[key="armeria.grpc.testing.SimpleRequest.NestedRequest", value=" A request nested in another request.
    "]]>
    but could not find:
     <[MapEntry[key="armeria.grpc.testing.SimpleRequest.NestedRequest", value=" A request nested in another request.
    "]]>
        at com.linecorp.armeria.server.grpc.GrpcDocStringExtractorTest.nestedMessage(GrpcDocStringExtractorTest.java:62)

com.linecorp.armeria.server.grpc.GrpcDocStringExtractorTest > nestedEnumType FAILED
    java.lang.AssertionError: 
    Expecting:
     <{}>
    to contain:
     <[MapEntry[key="armeria.grpc.testing.SimpleRequest.NestedEnum", value=" An enum nested in a request.
    "]]>
    but could not find:
     <[MapEntry[key="armeria.grpc.testing.SimpleRequest.NestedEnum", value=" An enum nested in a request.
    "]]>
        at com.linecorp.armeria.server.grpc.GrpcDocStringExtractorTest.nestedEnumType(GrpcDocStringExtractorTest.java:90)

com.linecorp.armeria.server.grpc.GrpcDocStringExtractorTest > field FAILED
    java.lang.AssertionError: 
    Expecting:
     <{}>
    to contain:
     <[MapEntry[key="armeria.grpc.testing.SimpleRequest/response_type", value=" Desired payload type in the response from the server.
     If response_type is RANDOM, server randomly chooses one from other formats.
    "]]>
    but could not find:
     <[MapEntry[key="armeria.grpc.testing.SimpleRequest/response_type", value=" Desired payload type in the response from the server.
     If response_type is RANDOM, server randomly chooses one from other formats.
    "]]>
        at com.linecorp.armeria.server.grpc.GrpcDocStringExtractorTest.field(GrpcDocStringExtractorTest.java:54)

com.linecorp.armeria.server.grpc.GrpcDocStringExtractorTest > nestedEnumValue FAILED
    java.lang.AssertionError: 
    Expecting:
     <{}>
    to contain:
     <[MapEntry[key="armeria.grpc.testing.SimpleRequest.NestedEnum/OK", value=" We're ok.
    "]]>
    but could not find:
     <[MapEntry[key="armeria.grpc.testing.SimpleRequest.NestedEnum/OK", value=" We're ok.
    "]]>
        at com.linecorp.armeria.server.grpc.GrpcDocStringExtractorTest.nestedEnumValue(GrpcDocStringExtractorTest.java:97)

com.linecorp.armeria.server.grpc.GrpcDocStringExtractorTest > enumValue FAILED
    java.lang.AssertionError: 
    Expecting:
     <{}>
    to contain:
     <[MapEntry[key="armeria.grpc.testing.CompressionType/NONE", value=" No compression
    "]]>
    but could not find:
     <[MapEntry[key="armeria.grpc.testing.CompressionType/NONE", value=" No compression
    "]]>
        at com.linecorp.armeria.server.grpc.GrpcDocStringExtractorTest.enumValue(GrpcDocStringExtractorTest.java:83)

com.linecorp.armeria.server.grpc.GrpcDocStringExtractorTest > message FAILED
    java.lang.AssertionError: 
    Expecting:
     <{}>
    to contain:
     <[MapEntry[key="armeria.grpc.testing.SimpleRequest", value=" Unary request.
    "]]>
    but could not find:
     <[MapEntry[key="armeria.grpc.testing.SimpleRequest", value=" Unary request.
    "]]>
        at com.linecorp.armeria.server.grpc.GrpcDocStringExtractorTest.message(GrpcDocStringExtractorTest.java:47)

com.linecorp.armeria.server.grpc.GrpcDocStringExtractorTest > nestedMessageField FAILED
    java.lang.AssertionError: 
    Expecting:
     <{}>
    to contain:
     <[MapEntry[key="armeria.grpc.testing.SimpleRequest.NestedRequest/nested_payload", value=" The payload for a nested request.
    "]]>
    but could not find:
     <[MapEntry[key="armeria.grpc.testing.SimpleRequest.NestedRequest/nested_payload", value=" The payload for a nested request.
    "]]>
        at com.linecorp.armeria.server.grpc.GrpcDocStringExtractorTest.nestedMessageField(GrpcDocStringExtractorTest.java:69)

com.linecorp.armeria.server.grpc.GrpcDocStringExtractorTest > service FAILED
    java.lang.AssertionError: 
    Expecting:
     <{}>
    to contain:
     <[MapEntry[key="armeria.grpc.testing.TestService", value=" A simple service to test the various types of RPCs and experiment with
     performance with various types of payload.
    "]]>
    but could not find:
     <[MapEntry[key="armeria.grpc.testing.TestService", value=" A simple service to test the various types of RPCs and experiment with
     performance with various types of payload.
    "]]>
        at com.linecorp.armeria.server.grpc.GrpcDocStringExtractorTest.service(GrpcDocStringExtractorTest.java:32)
```

--- a/grpc/build.gradle
+++ b/grpc/build.gradle
@@ -16,6 +16,17 @@ managedDependencies {
     }
 }
 
+def generatedFilesDir = "$projectDir/gen-src"
+
+sourceSets {
+    main {
+        output.dir("${generatedFilesDir}/main/resources", builtBy: 'generateProto')
+    }
+    test {
+        output.dir("${generatedFilesDir}/test/resources", builtBy: 'generateTestProto')
+    }
+}
+
 [tasks.shadedJar, tasks.shadedTestJar].each {
     it.relocate 'io.grpc.internal', "${shadedPackage}.grpc.internal"
 }
@@ -27,7 +38,7 @@ tasks.shadedJar.exclude {
 tasks.shadedJar.exclude 'META-INF/services/io.grpc.NameResolverProvider'
 
 protobuf {
-    generatedFilesBaseDir = "$projectDir/gen-src"
+    generatedFilesBaseDir = generatedFilesDir
     protoc {
         artifact = 'com.google.protobuf:protoc:3.2.0'
     }
@@ -47,7 +58,8 @@ protobuf {
             task.descriptorSetOptions.includeSourceInfo = true
             task.descriptorSetOptions.includeImports = true
             task.descriptorSetOptions.path =
-                    "${buildDir}/resources/main/META-INF/armeria/grpc/armeria-test.dsc"
+                    "${generatedFilesDir}/${task.sourceSet.name}/resources/" +
+                            "META-INF/armeria/grpc/armeria-${task.sourceSet.name}.dsc"
         }
     }
 }

